editor/components/colors/with-colors.js

```diff
@@ -1,91 +1,134 @@
 /**
  * External dependencies
  */
-import memoize from 'memize';
-import { get } from 'lodash';
+import { find, get, isFunction, isString, kebabCase, reduce, upperFirst } from 'lodash';
 
 /**
  * WordPress dependencies
  */
 import { createHigherOrderComponent, Component, compose } from '@wordpress/element';
 import { withSelect } from '@wordpress/data';
+import { deprecated } from '@wordpress/utils';
 
 /**
  * Internal dependencies
  */
-import { getColorValue, getColorClass, setColorValue } from './utils';
+import { getColorValue, getColorClass } from './utils';
+import { default as withColorsDeprecated } from './with-colors-deprecated';
 
 const DEFAULT_COLORS = [];
 
 /**
  * Higher-order component, which handles color logic for class generation
  * color value, retrieval and color attribute setting.
  *
- * @param {Function} mapGetSetColorToProps Function that receives getColor, setColor, and props,
- *                                         and returns additional props to pass to the component.
+ * @param {...(object|string)} args The arguments can be strings or objects. If the argument is an object,
+ *                                  it should contain the color attribute name as key and the color context as value.
+ *                                  If the argument is a string the value should be the color attribute name,
+ *                                  the color context is computed by applying a kebab case transform to the value.
+ *                                  Color context represents the context/place where the color is going to be used.
+ *                                  The class name of the color is generated using 'has' followed by the color name
+ *                                  and ending with the color context all in kebab case e.g: has-green-background-color.
+ *
  *
  * @return {Function} Higher-order component.
  */
-export default ( mapGetSetColorToProps ) => createHigherOrderComponent(
-	compose( [
-		withSelect(
-			( select ) => {
+export default ( ...args ) => {
+	if ( isFunction( args[ 0 ] ) ) {
+		deprecated( 'Using withColors( mapGetSetColorToProps ) ', {
+			version: '3.2',
+			alternative: 'withColors( colorAttributeName, { secondColorAttributeName: \'color-context\' }, ... )',
```

Comment:
It would be nicer to use double-quotes here rather than escaping the single-quotes inside. Is that against the style rules? (If so: ouch. 😅)